backend/src/config/commonFunctions.ts

```diff
@@ -0,0 +1,51 @@
+/*
+This file contains functions that are shared across the app
+ */
+
+/* istanbul ignore next */
+/**
+ * Returns the string that returns our NodeDescriptor object
+ * @param node Name of the variable in the query
+ */
+export function neo4jReturnNodeDescriptor(node: string): string {
+  return `ID(${node}) as id`;
```

Comment:
This code allows database injection attacks!

Reply:
this code should only be used in "backend"-generated queries.
`node` stands for the variable in the cypher query.
Example:
```ts
query = `MATCH (n) WHERE [...] RETURN ${neo4jReturnNodeDescriptor('n')}`; // variable is called n

query = `MATCH (a)-->(b) WHERE [...] RETURN ${neo4jReturnNodeDescriptor('b')}`; // cypher variables are called a and b
```

By using this approach we can use a common return statement in multiple queries that all will return NodeDescriptor without knowing exactly how the NodeDescriptor will look like.
This is more useful for returning "Node" or "Edge", since there are more parameters.
When we add an attribute to Node, then we would have to add that attribute to every cypher call that returns a Node-object. When we build the query using this function we have to add the new attribute once and it will modify all queries correctly.

It makes no sense to inject user generated strings here (and also should never be done).

Reply:
So are the parameter values known?  
Then add them in the function definition, instead of receiving arguments of type `string` and add a runtime check.

If the values are not known up-front, add a warning to the function comment, that the function MUST NOT be called with non-constant string arguments, due to possible database injection attacks.

The thing here is that currently, there is no attack possible, but if further development  in the backend does not use these functions with care, attacks may get possible then.